packages/cli/src/dioxus_crate.rs

```diff
@@ -139,10 +140,18 @@ pub struct DioxusCrate {
 }
 
 impl DioxusCrate {
+    // Maybe add optional LockOptions argument?
     pub fn new(target: &TargetArgs) -> Result<Self, CrateConfigError> {
+        // Use this struct in new/init subcommands?
         let mut cmd = Cmd::new();
+        cmd.lock_opts(LockOptions {
+            offline: true, // Don't access Internet when `dx clean` is run.
+            frozen: false,
+            locked: false,
+        });
         cmd.features(target.features.clone());
         let builder = krates::Builder::new();
+        // Tries to access Internet when offline.
         let krates = builder.build(cmd, |_| {})?;
         let package = find_main_package(target.package.clone(), &krates)?;
```

Comment:
Is this necessary here? I'm not sure `cargo clean` will attempt to access the internet, but I'm definitely not sure

Reply:
It wouldn't have been there if I didn't try calling `dx clean` on a place with no Internet connection. It was failing for some reason. That was the reason. I was thinking that this should be a separate issue. But I'm not sure. I think a global `--offline` (and others) flag should be added, and it should be enabled by default for `clean` subcommand.

So it's necessary if you are offline and want to just clean up the project dir. But an additional if statement is missing.

Reply:
Ohh ok that does sound like a great idea 👍 I’ll see if I can reproduce that error, that definitely sounds like the right fix